### PR TITLE
check for duplicate lids pointing at same file

### LINKED ISSUE #773: As a user, I want validate with the registry when a file is being referenced by more than one label
### Checked for duplicates

Yes - I've already checked

### 🧑‍🔬 User Persona(s)

Archivist

### 💪 Motivation

...so that I can use the registry to validate when more than one product references to the same file.

Related requirement:
- 🦄 #755

### 📖 Additional Details

_No response_

### Acceptance Criteria

**Given** a collection where >1 labels references the same file from `File_Area_*`
**When I perform** `validate-refs urn:pds:nasa:my_bundle::1.0`
**Then I expect** validate to throw an error

### ⚙️ Engineering Details

_No response_

### I&T
TestRail Test ID: T8681201

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Use opensearch to find all lids that reference same file.

## ⚙️ Test Data and/or Report
Had to use data from NASA-PDS/harvest#143. Created a single duplicate rqd. There are lots of missing secondary references and surprise duplicates:
```
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface - Reference Summary:
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -    22275 products processed
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -    115290 missing references
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -    0 fatals
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -    0 errors not including missing references
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -    0 warnings
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface - Duplicate Summary:
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -    Number of duplicates found: 2
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -    File: ps__0125_0678032243_000rqa__00417120483005510000___j01.csv
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -       referer: urn:nasa:pds:mars2020_pixl:data_oxides_pmc:ps__0125_0678032243_000rqa__00417120483005510000___j01
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -       referer: urn:nasa:pds:mars2020_pixl:data_oxides_pmc:ps__0125_0678032243_000rqd__00417120483005510000___j01
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -    File: readme.txt
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -       referer: urn:nasa:pds:mars2020.spice
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -       referer: urn:nasa:pds:insight_rad
15:45:14.943 [main] INFO  gov.nasa.pds.validate.ri.CommandLineInterface -       referer: urn:nasa:pds:mars2020_pixl
```

## ♻️ Related Issues
Closes #773

### CODE REVIEW

**Comment by jordanpadams** [comment]
@al-niessner I think we need to use `ops:Data_File_Info.ops:file_ref` not `ops:Data_File_Info.ops:file_name` or `pds:File.pds:file_name`, because as you noted, the path where that file exists matters. There are going to be tons of the same `file_name` values in the system, but if they exist in a different bundle somewhere, it is not a problem.

File: src/main/java/gov/nasa/pds/validate/ri/CommandLineInterface.java [hunk 1]
```diff
@@ -18,18 +20,17 @@
 
 public class CommandLineInterface {
   final private Logger log = LogManager.getLogger(CommandLineInterface.class);
-  private long broken = -1, total = -1;
   final private Options opts;
+  private long broken = -1, total = -1;
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/validate/ri/CommandLineInterface.java:24]: I think my old type-unsafe C days are kicking in. I know Java's doing the right thing, but part of me wants to see
```java
private long broken = -1L, total = -1L;
```
No change needed; just reminiscing 😇

File: src/main/java/gov/nasa/pds/validate/ri/CommandLineInterface.java [hunk 2]
```diff
@@ -18,18 +20,17 @@
...
+  private Map<String, List<String>> duplicates = null;
 
   public CommandLineInterface() {
     super();
     this.opts = new Options();
-    
-    // Disabling this argument for the time being since the Search API does not yet support authorized access
+    /* registry is not ready. Deactivating this option until it becomes ready
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/validate/ri/CommandLineInterface.java:30]: Thank you, thank you, **thank you** for including a comment explaining _why_ something is being commented-out! 🎉

**Review by nutjob4life — APPROVED** [review]
👍

**Comment by jordanpadams** [comment]
@al-niessner how does the testing check out using `file_ref` vs. `file_name`?

**Comment by jordanpadams** [comment]
See https://github.com/NASA-PDS/validate/issues/773#issuecomment-1850748550 for example test run

**Pull request merged**